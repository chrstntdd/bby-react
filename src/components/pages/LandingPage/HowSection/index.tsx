import * as React from 'react';
import './how-section.scss';
import Card from '../HowCard';

const HowSection = props => {
  const howCards = props.cardData.map((card, index) => <Card key={index} {...card} />);
  return (
    <section id="how-section">
      <div className="card-container">{howCards}</div>
    </section>
  );
};

export default HowSection;
