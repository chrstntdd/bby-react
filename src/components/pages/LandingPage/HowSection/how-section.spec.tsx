import * as React from 'react';
import { shallow } from 'enzyme';

import HowSection from './';

describe('<HowSection/>', () => {
  it('should render without crashing', () => {
    const props = {
      cardData: [
        {
          name: 'simple',
          img: 'fakeimg.jpg',
          desc: '...but powerful'
        },
        {
          name: 'reliable',
          img: 'fakeimg.jpb',
          desc: 'There when you need it'
        },
        {
          name: 'safe',
          img: 'fakeimg.jpg',
          desc: 'Without compromises'
        },
        {
          name: 'fast',
          img: 'fakeimg.jpg',
          desc: 'Designed with speed in mind'
        }
      ]
    };
    shallow(<HowSection {...props} />);
  });
});
