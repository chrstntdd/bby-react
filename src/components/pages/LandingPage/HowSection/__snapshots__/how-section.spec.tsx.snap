// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HowSection/> should render without crashing 1`] = `
<section
  id="how-section"
>
  <div
    className="card-container"
  >
    <Card
      desc="...but powerful"
      img="fakeimg.jpg"
      key="0"
      name="simple"
    />
    <Card
      desc="There when you need it"
      img="fakeimg.jpb"
      key="1"
      name="reliable"
    />
    <Card
      desc="Without compromises"
      img="fakeimg.jpg"
      key="2"
      name="safe"
    />
    <Card
      desc="Designed with speed in mind"
      img="fakeimg.jpg"
      key="3"
      name="fast"
    />
  </div>
</section>
`;
